ansible/site.yml

```diff
@@ -20,131 +13,173 @@
+    # commented out above where the ec2 instances are enumerated.
+    - name: Create inventory group 'client'
+      add_host: hostname={{ item }} groupname=client
+      with_items: jmeter-client 
     
-    - name: EC2 - Provision Security Group 'ssh-all'
-      ec2_group:
-        name: ssh-all
-        description: Allows incoming traffic on port 22 (from all IPs).
-        region: "{{ region }}"
-        rules:
-          - proto: tcp
-            from_port: 22
-            to_port: 22
-            cidr_ip: 0.0.0.0/0
-        rules_egress:
-          - proto: all
-            cidr_ip: 0.0.0.0/0
-    
-    - name: EC2 - Provision 'bluebutton-stress' master
-      ec2:
-        key_name: "{{ ec2_key_name }}"
-        group:
-          - default
-          - ssh-all
-        instance_type: "{{ ec2_instance_type }}"
-        image: "{{ ami_id_ubuntu }}"
-        region: "{{ region }}"
-        wait: true
-        exact_count: 1
-        count_tag:
-          Name: bluebutton-stress-master
-          Application: "BlueButton"
-        instance_tags:
-          Name: bluebutton-stress-master
-          Application: "BlueButton"
-          CreatedBy: "{{ whoami.stdout }}"
-      register: ec2_master
-    
-    - name: EC2 - Provision 'bluebutton-stress' slaves
-      ec2:
-        key_name: "{{ ec2_key_name }}"
-        group:
-          - default
-          - ssh-all
-        instance_type: "{{ ec2_instance_type }}"
-        image: "{{ ami_id_ubuntu }}"
-        region: "{{ region }}"
-        wait: true
-        # The default of 300s here wasn't cutting it for 8 slaves.
-        wait_timeout: 600
-        exact_count: "{{ num_slaves }}"
-        count_tag:
-          Name: bluebutton-stress-slave
-          Application: "BlueButton"
-        instance_tags:
-          Name: bluebutton-stress-slave
-          Application: "BlueButton"
-          CreatedBy: "{{ whoami.stdout }}"
-      register: ec2_slaves
-    
-    - name: Create inventory group 'master'
-      add_host: hostname={{ item.public_ip }} groupname=master
-      with_items: ec2_master.instances
-    
-    - name: Create inventory group 'slaves'
-      add_host: hostname={{ item.public_ip }} groupname=slaves
-      with_items: ec2_slaves.instances
+    - name: Create inventory group 'servers'
+      add_host: hostname={{ item }} groupname=servers
+      with_items: jmeter-server 
     
-    - name: Wait for SSH to come up
-      wait_for: host={{ item.public_dns_name }} port=22 timeout=320 state=started
-      with_flattened:
-        - ec2_master.instances
-        - ec2_slaves.instances
-      
-- name: Configure instance(s)
-  hosts: master:slaves
+#    - name: Wait for SSH to come up
+#      wait_for: host={{ item.public_dns_name }} port=22 timeout=320 state=started
+#      with_flattened:
+#        - ec2_client.instances
+#        - ec2_servers.instances
+
+- name: Configure roles common to all servers 
+  hosts: client:servers
```

Comment:
Why is `client` singular and `servers` plural? Does JMeter name things backwards or is that just an oops?

Reply:
Yeah, in jmeter nomenclature the client(test controller) is the "master" and the servers(test runners) are the "slaves".  See here: http://jmeter.apache.org/usermanual/remote-test.html